src/overlays/actors/ovl_En_Bigpamet/z_en_bigpamet.c

```diff
@@ -90,121 +113,757 @@ static DamageTable D_80A29734 = {
+        func_80A28970(this);
+    }
+}
+
+void func_80A28970(EnBigpamet* this) {
+    Actor_PlaySfxAtPos(&this->actor, NA_SE_EN_PAMET_CUTTER_OFF);
+    this->actor.shape.rot.z = 0;
+    this->collider.base.atFlags &= ~AT_ON;
+    this->collider.info.bumper.dmgFlags = 0xF7CFFFFF;
+    this->actor.speedXZ = 0.0f;
+    this->actionFunc = func_80A289C8;
+}
+
+void func_80A289C8(EnBigpamet* this, PlayState* play) {
+    this->unk_2A8 -= 0.1f;
+    if (this->unk_2A8 < 0.5f) {
+        func_80A28A28(this);
+    }
+    func_80A27DD8(this, play);
+}
+
+void func_80A28A28(EnBigpamet* this) {
+    this->actor.draw = EnBigpamet_Draw;
+    Animation_MorphToPlayOnce(&this->skelAnime2, &object_tl_Anim_0031DC, -3.0f);
+    this->unk_2A8 = 0.1f;
+    this->unk_2A4 = 1.0f;
+    this->actor.world.rot.y = this->actor.shape.rot.y;
+    this->actionFunc = func_80A28A98;
+}
+
+void func_80A28A98(EnBigpamet* this, PlayState* play) {
+    f32 curFrame;
+
+    Math_ScaledStepToS(&this->actor.shape.rot.y, this->actor.yawTowardsPlayer, 0x600);
+
+    if (SkelAnime_Update(&this->skelAnime2)) {
+        if ((this->actor.parent->params == 7) || (this->actor.parent->params == 8)) {
+            func_80A283F0(this);
+        } else {
+            func_80A284E4(this);
+        }
+    } else {
+        curFrame = this->skelAnime2.curFrame;
+
+        if (curFrame > 7.0f) {
+            this->unk_2A8 = 1.5f - ((curFrame - 7.0f) * (1.0f / 6.0f));
+            this->unk_2A4 = 1.5f - ((curFrame - 7.0f) * (1.0f / 6.0f));
+        } else {
+            this->unk_2A8 = (0.2f * curFrame) + 0.1f;
+            this->unk_2A4 = ((1.0f / 14.0f) * curFrame) + 1.0f;
+        }
+    }
+}
+
+void func_80A28B98(EnBigpamet* this, PlayState* play) {
+    Actor* sp24;
+
+    Animation_PlayLoop(&this->skelAnime2, &object_tl_Anim_000AF4);
+
+    this->collider.base.atFlags &= ~AT_ON;
+    this->collider.info.bumper.dmgFlags = 0xF7CFFFFF;
+    this->collider.base.atFlags &= ~(AT_HIT | AT_BOUNCED);
+    this->collider.base.acFlags &= ~AC_ON;
+
+    this->actor.velocity.y = 22.0f;
+    this->actor.speedXZ = 5.0f;
+
+    if ((this->actor.draw == func_80A2966C) && (this->actionFunc != func_80A28DC0)) {
+        this->actor.draw = EnBigpamet_Draw;
+    } else if (this->collider.base.ac != NULL) {
+        this->actor.world.rot.y = BINANG_ROT180(Actor_YawBetweenActors(&this->actor, this->collider.base.ac));
+    }
+
+    this->actor.shape.rot.y = this->actor.world.rot.y;
+    this->actor.bgCheckFlags &= ~1;
+    this->actor.flags &= ~ACTOR_FLAG_1;
+    this->actor.params = ENBIGPAMET_0;
+
+    if ((this->actor.parent->params == 3) || (this->actor.parent->params == 4) || (this->actor.parent->params == 8)) {
+        this->actor.parent->params = 5;
+    }
+
+    sp24 = Item_DropCollectible(play, &this->actor.world.pos, ITEM00_ARROWS_10);
+    if (sp24 != NULL) {
+        sp24->velocity.y = 15.0f;
+        sp24->world.rot.y = Actor_YawToPoint(&this->actor, &this->actor.home.pos);
+    }
```

Comment:
```suggestion
    collectible = Item_DropCollectible(play, &this->actor.world.pos, ITEM00_ARROWS_10);
    if (collectible != NULL) {
        collectible->velocity.y = 15.0f;
        collectible->world.rot.y = Actor_YawToPoint(&this->actor, &this->actor.home.pos);
    }
```
This is a standard name for this function